pkg/registry/pod/etcd/etcd.go

```diff
@@ -218,6 +220,38 @@ type LogREST struct {
 	kubeletConn client.ConnectionInfoGetter
 }
 
+// Max length read from the body of a log location which returns error status
+const (
+	maxReadLength = 1000
+)
+
+// A log location response checker to transform the error.
+type LogLocationResponseChecker struct {
+	podName string
```

Comment:
Sure - probably be good to make sure the Godoc indicates this is a generic
HTTP check, vs a specific one so someone coming along later doesn't try to
get to clever.

On Wed, Sep 23, 2015 at 6:25 PM, Daniel Smith <notifications@github.com>
wrote:

> In pkg/registry/pod/etcd/etcd.go
> <https://github.com/kubernetes/kubernetes/pull/13722#discussion_r40265907>
> :
>
> > @@ -218,6 +220,38 @@ type LogREST struct {
> >  	kubeletConn client.ConnectionInfoGetter
> >  }
> >
> > +// Max length read from the body of a log location which returns error status
> > +const (
> > +	maxReadLength = 1000
> > +)
> > +
> > +// A log location response checker to transform the error.
> > +type LogLocationResponseChecker struct {
> > +	podName string
>
> But looking at the logic of the checker, everything is standard and
> applicable to basically any HTTP call. If we mutate the kubelet api, we can
> always copy this back.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13722/files#r40265907>.
>